hr_employee_language/models/hr_employee_language.py

```diff
@@ -1,19 +1,44 @@
 # Copyright (C) 2017-Today: Odoo Community Association (OCA)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import fields, models, tools
+from odoo import _, fields, models
 
 
 class HrEmployeeLanguage(models.Model):
     _name = "hr.employee.language"
     _description = "HR Employee Language"
 
-    name = fields.Selection(
-        tools.misc.scan_languages(), string="Language", required=True
-    )
+    def _get_selection(self):
+        """
+        generate language selection from available on db
+        """
+        selection = []
+        for lang in self.env["res.lang"].search([("active", "in", (True, False))]):
+            selection += [
+                (_("%(code)s", code=lang.code), _("%(name)s", name=lang.name))
+            ]
+        return selection
+
+    name = fields.Selection(_get_selection, string="Language", required=True)
     description = fields.Char()
     employee_id = fields.Many2one("hr.employee", string="Employee", required=True)
     can_read = fields.Boolean(string="Read", default=True)
     can_write = fields.Boolean(string="Write", default=True)
     can_speak = fields.Boolean(string="Speak", default=True)
     can_listen = fields.Boolean(string="Listen", default=True)
+
+    def _compute_display_name(self):
+        """
+        redefine record name to show languages name instead languages code
+        """
+        result = []
+        all_lang = self.env["res.lang"].search([("active", "in", (True, False))])
+        self_data = self if self else all_lang
+        for record in self_data:
+            lang = self.env["res.lang"].search(
+                [("code", "=", record.name), ("active", "in", (True, False))], limit=1
+            )
+            language = lang.name if lang else record.name
+            name = _("%(lang)s", lang=language)
+            result.append((record.id, name))
```

Comment:
isn't the display_name calculation assignment?
```suggestion
            record.display_name = lang.name if lang else record.name
```